nixos/modules/services/networking/knot.nix

```diff
@@ -60,11 +160,21 @@ in {
         '';
       };
 
-      extraConfig = mkOption {
-        type = types.lines;
-        default = "";
+      settings = mkOption {
+        type = types.attrs;
```

Comment:
Thanks for this huge improvement! Please be aware that this `type.attrs` does not merge the `settings` as expected with other options, especially when the configuation is split between modules. Using a freeform submodule as suggested in the [RFC followed by this PR](https://github.com/NixOS/rfcs/blob/master/rfcs/0042-config-option.md#part-1-1) would solve that.

Reply:
That should be in PR #304373